### PR TITLE
📖  Document code-cell directive and cell tag functionality

### PR DESCRIPTION
**Pull request by choldgraf**
This makes several documentation improvements around our code cell behavior:

### Document `{code-cells}`

I noticed that there's no documentation of the `{code-cell}` directive for executing content within markdown files. From looking at the code it sounds like this is still an alpha-level feature, but I thought it'd be helpful to document it since the functionality is technically there, even though it still has a ways to go.

### Document `{eval}`

This also documents our inline expressions with `{eval}` a bit more clearly, and so closes:

- closes https://github.com/executablebooks/mystmd/issues/1109

### Documenting tags to control code cells

This also adds some documentation for the supported tags to control cell behavior. While I was documenting that, I also uncovered this bug:

- #1112 

And think we should consider executing content with our docs here:

- https://github.com/executablebooks/mystmd/issues/1114

### Making some titles more precise

While making these changes I felt like the "short titles" for some of the pages were ambiguous because of their use of overloaded terms (e.g. `Executing Notebooks` can mean many things). I tried to rename to make things clearer.

### Clarified the manual jupyter server instructions

I tried to make the jupyter server instructions easier to follow for manually-specifying the server for execution.

### CODE REVIEW

File: docs/notebooks-with-markdown.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Jupyter cells with Markdown files
+
+You can specify Jupyter code cells in your markdown, which allows you to execute computation using [MyST's notebook execution engine](./execute-notebooks.md).
+
+```{warning} This is a beta feature
```
Comment (choldgraf) [inline docs/notebooks-with-markdown.md:5]:
```suggestion
```{warning} This is an alpha feature
```

**Comment by choldgraf** [comment]
Hold off a sec on this one, I didn't realize that `{eval}` was already supported. I'm going to document that quickly so that we can close

- https://github.com/executablebooks/mystmd/issues/1109

File: docs/execute-notebooks.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ---
-title: Execute Notebooks with Jupyter
+title: Execute Notebooks during your build
```
Comment (choldgraf) [inline docs/execute-notebooks.md:2]:
```suggestion
title: Execute Notebooks During Your Build
```

**Comment by choldgraf** [comment]
OK I think this one is ready to go now. Requesting a review/merge from @agoose77 since this touches on some of the eval and execution work you did

File: docs/execute-notebooks.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 thumbnail: thumbnails/execute-notebooks.png
 ---
 
-## Overview
-
 :::{warning} MyST Execution is in Beta
-We are adding support for executing markdown notebooks and ipynb files, including inline execution. As we are adding this functionality we appreciate any feedback from the community on how it is working in your environments. Please add [issues](https://github.com/executablebooks/mystmd/issues/new) or join [Discord](https://discord.mystmd.org/) to give feedback.
+By default, execution is disabled and code outputs are only inserted if the notebook has already been executed (for text-based notebooks, there are no outputs).
+We are adding support for executing markdown notebooks and ipynb files, including inline execution.
+As we are adding this functionality we appreciate any feedback from the community on how it is working in your environments. Please add [issues](https://github.com/executablebooks/MyST/issues/new) or join [Discord](https://discord.MyST.org/) to give feedback.
```
Comment (rowanc1) [inline docs/execute-notebooks.md:11]:
```suggestion
As we are adding this functionality we appreciate any feedback from the community on how it is working in your environments. Please add [issues](https://github.com/executablebooks/mystmd/issues/new) or join [Discord](https://discord.mystmd.org/) to give feedback.
```

Comment (choldgraf) [reply docs/execute-notebooks.md:11]: ahhh that explains why I got diff conflicts haha, sorry about that I didn't notice the change

**Comment by choldgraf** [comment]
Also just noting that I don't necessarily want to turn @agoose77 into a _bottleneck_ for merging, if @rowanc1 has the info to know that our `{eval}` description is correct, then I'd say just go for it. I just pinged @agoose77 since I know he worked on it, but my general _modus operandi_ is bias towards action unless explicitly stated otherwise :-D

File: docs/execute-notebooks.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
 ---
-title: Execute Notebooks with Jupyter
+title: Execute Notebooks During Your Build
 description: MyST can execute notebooks using Jupyter Server, making it possible to build rich websites and documents from text-based notebooks.
-short_title: Executing Notebooks
+short_title: Execute Notebooks During Your Build
```
Comment (agoose77) [inline docs/execute-notebooks.md:4]: @rowanc1 this should probably be shorter as it's part of the sidebar nav?

File: docs/execute-notebooks.md [hunk 5]
```diff
@@ -18,11 +18,41 @@ myst start --execute
...
+```
+
+Alternatively, you can manually delete the `execute/` folder in your build folder, e.g.:
+
+```bash
+rm -rf _build/execute
+```
 
-## Launching a Server
+## How MyST executes your code
```
Comment (agoose77) [inline docs/execute-notebooks.md:52]: @choldgraf love this change

File: docs/execute-notebooks.md (hunk 4 above)
Comment (rowanc1) [reply docs/execute-notebooks.md:4]: Yes, probably. :)

**Pull request merged**